Brokerages/Bitfinex/BitfinexBrokerage.cs

```diff
@@ -0,0 +1,449 @@
+        /// <returns></returns>
+        public override List<Order> GetOpenOrders()
+        {
+
+            var list = this.GetOpenBitfinexOrders().Select(o => (Order)o).ToList();
+
+            foreach (var item in list)
+            {
+                if (item.Status != OrderStatus.Canceled && item.Status != OrderStatus.Filled && item.Status != OrderStatus.Invalid)
+                {
+                    var cached = this.CachedOrderIDs.Where(c => c.Value.BrokerId.Contains(item.BrokerId.First()));
+                    if (cached.Count() > 0 && cached.First().Value != null)
+                    {
+                        this.CachedOrderIDs[cached.First().Key] = item;
+                    }
+                }
+            }
+            return list;
+        }
+
+        /// <summary>
+        /// Retreive holdings from exchange
+        /// </summary>
+        /// <returns></returns>
+        public override List<Holding> GetAccountHoldings()
+        {
+
+            var list = new List<Holding>();
+
+            var response = _restClient.GetActivePositions();
+            foreach (var item in response)
+            {
+                var ticker = _restClient.GetPublicTicker(TradingApi.ModelObjects.BtcInfo.PairTypeEnum.btcusd, TradingApi.ModelObjects.BtcInfo.BitfinexUnauthenicatedCallsEnum.pubticker);
+                list.Add(new Holding
+                {
+                    Symbol = Symbol.Create(item.Symbol, SecurityType.Forex, Market.Bitfinex),
+                    Quantity = decimal.Parse(item.Amount) * ScaleFactor,
+                    Type = SecurityType.Forex,
+                    CurrencySymbol = "Ƀ",
+                    ConversionRate = (decimal.Parse(ticker.Mid) / ScaleFactor),
+                    MarketPrice = (decimal.Parse(ticker.Mid) / ScaleFactor),
+                    AveragePrice = (decimal.Parse(item.Base) / ScaleFactor),
+                });
+            }
+            return list;
+        }
+
+
+        /// <summary>
+        /// Get Cash Balances from exchange
+        /// </summary>
+        /// <returns></returns>
+        //todo: handle other currencies
+        public override List<Securities.Cash> GetCashBalance()
+        {
+            var list = new List<Securities.Cash>();
+            var response = _restClient.GetBalances();
+            foreach (var item in response)
+            {
+                if (item.Type == Wallet)
+                {
+                    if (item.Currency == "usd")
+                    {
+                        list.Add(new Securities.Cash(item.Currency, item.Amount, 1));
+                    }
+                    else
+                    {
+                        var ticker = _restClient.GetPublicTicker(TradingApi.ModelObjects.BtcInfo.PairTypeEnum.btcusd, TradingApi.ModelObjects.BtcInfo.BitfinexUnauthenicatedCallsEnum.pubticker);
+                        list.Add(new Securities.Cash("BTC", item.Amount * ScaleFactor, decimal.Parse(ticker.Mid) / ScaleFactor));
```

Comment:
Instead of explicitly stating 'BTC', maybe using `item.Currency` is more appropriate. Also, similar comment as above, can we resolve the `ticker` using `item.Currency` instead of hard coding it for 'BTCUSD'?